### PR TITLE
Switch from using bash scripts to python scripts for creating resources

### PR DESCRIPTION
**Pull request by maxdymond**
This pull request includes multiple changes related to improving code quality, updating documentation, and modifying the build process. The most significant changes include updating the Python version requirement, removing shell scripts in favor of Python scripts, and modifying the linting and static analysis tools used in the project.

Changes to Python code and build process:

* [`.flake8`](diffhunk://#diff-6951dbb399883798a226c1fb496fdb4183b1ab48865e75eddecf6ceb6cf46442R4-R34): Added rules to exclude certain directories and files from flake8 checking and added rules for flake8 to ignore certain warnings and errors.
* [`.github/workflows/ci.yml`](diffhunk://#diff-b803fcb7f17ed9235f1e5cb1fcd2f5d3b2838429d4368ae4c57ce4436577f03fL21-R26): Removed the step for running pylint and added a step for running flake8.
* [`pyproject.toml`](diffhunk://#diff-50c86b7ed8ac2cf95bd48334961bf0530cdc77b5a56f852c5c61b89d735fd711L10-R28): Updated the Python version requirement to 3.8.2, removed pylint from dev dependencies, and added flake8 and its plugins. Also added a script for creating resources in Azure.
* [`function_app.py`](diffhunk://#diff-e2c0b572ad19f63b34e908d655a1c7a049e1b1fcc8171ebc48ad3b009b6b5b03L15-R16): Removed an unnecessary import statement.
* `create_resources.sh` and `create_resources_nosharedkey.sh`: Both shell scripts were removed. [[1]](diffhunk://#diff-94a13fe901b3168471735baeb7c32da918d2f4d71a095a35e1b89117fb2b5ad9L1-L125) [[2]](diffhunk://#diff-bdaf542109c7aa590311ce3b1c0528088a66362fb4c9812e1abc6067bf922186L1-L147)
* `src/apt_package_function/__init__.py`, `src/apt_package_function/azcmd.py`, and `src/apt_package_function/bicep_deployment.py`: Added new Python scripts to replace the functionality of the removed shell scripts. [[1]](diffhunk://#diff-a2ce4ed24860ff844cb00e6cdead6782c654eed326da9e08e0862a9ebf2c82d8R1-R87) [[2]](diffhunk://#diff-c1c4dfa84c8c082c8185bc01f05289a2d36eb4752a5a78eee0d5335c8d6d63c3L139-R139) [[3]](diffhunk://#diff-b9b03eeaf7d241e4a359b893da5c4fe45316b405a5617be0ba6969517a1e9453R1-R89)

Changes to documentation:

* [`README.md`](diffhunk://#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5R9-R47): Updated the documentation to reflect the changes in the build process and the usage of the project. Also improved the explanation of the speed of repository updates. [[1]](diffhunk://#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5R9-R47) [[2]](diffhunk://#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5L45-R60) [[3]](diffhunk://#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5L68-R89)

Changes to Bicep deployment:

* [`rg.bicep`](diffhunk://#diff-c1c4dfa84c8c082c8185bc01f05289a2d36eb4752a5a78eee0d5335c8d6d63c3L139-R139): Modified the output of the function app name.

### CODE REVIEW

File: src/apt_package_function/__init__.py [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+        log_filename = name
+
+    formatter = logging.Formatter("%(asctime)s %(levelname)-5.5s %(message)s")
+
+    log_path = Path(log_filename)
+
+    # Get the current time as a timestamp
+    timestamp = datetime.now().strftime("%Y-%m-%d_%H-%M-%S")
+
+    sojournerdir = Path(__file__).parent.parent.parent
```
Comment (tofay) [inline src/apt_package_function/__init__.py:27]: why is this called sojournerdir?

File: src/apt_package_function/func_app.py [hunk 2]
```diff
@@ -0,0 +1,198 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+"""Management of function applications"""
```
Comment (tofay) [inline src/apt_package_function/func_app.py:3]: did you consider using azure sdk and /or azure-functions library?

**Review by tofay — APPROVED** [review]

File: src/apt_package_function/__init__.py (hunk 1 above)
Comment (maxdymond) [reply src/apt_package_function/__init__.py:27]: Good question, should be rootdir really.

File: src/apt_package_function/func_app.py (hunk 2 above)
Comment (maxdymond) [reply src/apt_package_function/func_app.py:3]: I did but pulling in the Azure SDK in pure Python is several hundreds of megabytes and dependencies. I might be able to use the management libraries more directly but this is a good stopgap.

**Pull request merged**